### PR TITLE
Align help message with default values

### LINKED ISSUE #2552: Align help message with default values
When looking at the help message for the collector, some flag descriptions are a bit confusing:

```
--admin.http.host-port string                    The host:port (e.g. 127.0.0.1:5555 or :5555) for the admin server, including health check, /metrics, etc. (default ":14269")

--collector.http-server.host-port string         The host:port (e.g. 127.0.0.1:9411 or :9411) of the collector's HTTP server (default ":14268")

--collector.zipkin.host-port string              The host:port (e.g. 127.0.0.1:5555 or :5555) of the collector's Zipkin server (default ":0")
```

I would expect to see something like this:

```
--admin.http.host-port string                    The host:port (e.g. 127.0.0.1:14269 or :14269) for the admin server, including health check, /metrics, etc. (default ":14269")

--collector.http-server.host-port string         The host:port (e.g. 127.0.0.1:14268 or :14268) of the collector's HTTP server (default ":14268")

--collector.zipkin.host-port string              The host:port (e.g. 127.0.0.1:9411 or :9411) of the collector's Zipkin server (default ":0")
```

This might also be happening with other ports.

### PR DESCRIPTION
**Pull request by tobiaskohlbau**
Align ports included in help messages with the jaeger default
ports specified in ports package.

Signed-off-by: Tobias Kohlbau <tobias@kohlbau.de>

## Which problem is this PR solving?
- Resolves #2552 

## Short description of the changes
- Aligning the ports shown in help messages with default ports defined in ports package.

### CODE REVIEW

File: cmd/collector/app/builder_flags.go [hunk 1]
```diff
@@ -94,14 +94,14 @@ func AddFlags(flags *flag.FlagSet) {
...
-	flags.String(CollectorHTTPHostPort, ports.PortToHostPort(ports.CollectorHTTP), "The host:port (e.g. 127.0.0.1:9411 or :9411) of the collector's HTTP server")
+	flags.String(CollectorHTTPHostPort, ports.PortToHostPort(ports.CollectorHTTP), "The host:port (e.g. 127.0.0.1:14268 or :14268) of the collector's HTTP server")
 	flags.String(CollectorGRPCHostPort, ports.PortToHostPort(ports.CollectorGRPC), "The host:port (e.g. 127.0.0.1:14250 or :14250) of the collector's GRPC server")
 	tlsFlagsConfig.AddFlags(flags)
 }
 
 // AddOTELZipkinFlags adds flag that are exposed by OTEL Zipkin receiver
 func AddOTELZipkinFlags(flags *flag.FlagSet) {
-	flags.String(CollectorZipkinHTTPHostPort, ports.PortToHostPort(0), "The host:port (e.g. 127.0.0.1:5555 or :5555) of the collector's Zipkin server")
+	flags.String(CollectorZipkinHTTPHostPort, ports.PortToHostPort(0), "The host:port (e.g. 127.0.0.1:0 or :0) of the collector's Zipkin server")
```
Comment (tobiaskohlbau) [inline cmd/collector/app/builder_flags.go:104]: I'm not sure if _127.0.0.1:0 or :0_ is an advantage. Maybe it's an option to state that _:0_ means a random port?

Comment (jpkrohling) [reply cmd/collector/app/builder_flags.go:104]: This one is an exception: it should be 9411 in the example, which is the standard Zipkin port even though we use 0, as we don't open it by default unles it's explicitly set.

File: cmd/flags/admin.go [hunk 2]
```diff
@@ -75,7 +75,7 @@ func (s *AdminServer) setLogger(logger *zap.Logger) {
 func (s *AdminServer) AddFlags(flagSet *flag.FlagSet) {
 	flagSet.Int(healthCheckHTTPPort, 0, healthCheckHTTPPortWarning+" see --"+adminHTTPHostPort)
 	flagSet.Int(adminHTTPPort, 0, adminHTTPPortWarning+" see --"+adminHTTPHostPort)
-	flagSet.String(adminHTTPHostPort, s.adminHostPort, "The host:port (e.g. 127.0.0.1:5555 or :5555) for the admin server, including health check, /metrics, etc.")
+	flagSet.String(adminHTTPHostPort, s.adminHostPort, "The host:port (e.g. 127.0.0.1:14269 or :14269) for the admin server, including health check, /metrics, etc.")
```
Comment (jpkrohling) [inline cmd/flags/admin.go:78]: This one might be tricky, as the admin port is used by multiple places with different values:

```console
$ podman run -e SPAN_STORAGE_TYPE=memory docker.io/jaegertracing/jaeger-ingester:latest --help | grep "admin.http.host-port string"
      --admin.http.host-port string                   The host:port (e.g. 127.0.0.1:5555 or :5555) for the admin server, including health check, /metrics, etc. (default ":14270")
```

Perhaps it should use `fmt.Sprintf("The host:port (e.g. 127.0.0.1:%s...", s.adminHostPort, ...)`

**Review by jpkrohling — COMMENTED** [review]
Almost there! There's something specific to the admin port that could be handled as well, but already looks better than before!

File: cmd/flags/admin.go (hunk 2 above)
Comment (tobiaskohlbau) [reply cmd/flags/admin.go:78]: Good catch! I've updated the code.

File: cmd/collector/app/builder_flags.go (hunk 1 above)
Comment (tobiaskohlbau) [reply cmd/collector/app/builder_flags.go]: I've addressed this exception.

File: cmd/flags/admin.go [hunk 3]
```diff
@@ -75,7 +76,7 @@ func (s *AdminServer) setLogger(logger *zap.Logger) {
 func (s *AdminServer) AddFlags(flagSet *flag.FlagSet) {
 	flagSet.Int(healthCheckHTTPPort, 0, healthCheckHTTPPortWarning+" see --"+adminHTTPHostPort)
 	flagSet.Int(adminHTTPPort, 0, adminHTTPPortWarning+" see --"+adminHTTPHostPort)
-	flagSet.String(adminHTTPHostPort, s.adminHostPort, "The host:port (e.g. 127.0.0.1:5555 or :5555) for the admin server, including health check, /metrics, etc.")
+	flagSet.String(adminHTTPHostPort, s.adminHostPort, fmt.Sprintf("The host:port (e.g. 127.0.0.1:%s or :%s) for the admin server, including health check, /metrics, etc.", s.adminHostPort))
```
Comment (jpkrohling) [inline cmd/flags/admin.go:79]: Looks like it's missing the value for the second parameter:

```
      --admin.http.host-port string                     The host:port (e.g. 127.0.0.1::14269 or :%!s(MISSING)) for the admin server, including health check, /metrics, etc. (default ":14269")
```

Comment (tobiaskohlbau) [reply cmd/flags/admin.go:79]: Of course it needs to be there. Sorry for bothering you with such stupid mistakes. I've run your command and checked all outputs for now. It seems that `s.adminHostPort` already contains a colon therefore I updated the sourcecode to reflect that changes.

**Comment by yurishkuro** [comment]
Tip: use `Resolves #2552 ` instead of just `#2552` when referring to a ticket, so that the ticket is auto-closed by GitHub once the PR is merged. I updated the description accordingly.

File: cmd/collector/app/builder_flags.go [hunk 4]
```diff
@@ -94,14 +94,14 @@ func AddFlags(flags *flag.FlagSet) {
...
-	flags.String(CollectorHTTPHostPort, ports.PortToHostPort(ports.CollectorHTTP), "The host:port (e.g. 127.0.0.1:9411 or :9411) of the collector's HTTP server")
+	flags.String(CollectorHTTPHostPort, ports.PortToHostPort(ports.CollectorHTTP), "The host:port (e.g. 127.0.0.1:14268 or :14268) of the collector's HTTP server")
 	flags.String(CollectorGRPCHostPort, ports.PortToHostPort(ports.CollectorGRPC), "The host:port (e.g. 127.0.0.1:14250 or :14250) of the collector's GRPC server")
 	tlsFlagsConfig.AddFlags(flags)
 }
 
 // AddOTELZipkinFlags adds flag that are exposed by OTEL Zipkin receiver
 func AddOTELZipkinFlags(flags *flag.FlagSet) {
-	flags.String(CollectorZipkinHTTPHostPort, ports.PortToHostPort(0), "The host:port (e.g. 127.0.0.1:5555 or :5555) of the collector's Zipkin server")
+	flags.String(CollectorZipkinHTTPHostPort, ports.PortToHostPort(0), "The host:port (e.g. 127.0.0.1:9411 or :9411) of the collector's Zipkin server")
```
Comment (yurishkuro) [inline cmd/collector/app/builder_flags.go:104]: shouldn't this default to empty string?
```suggestion
	flags.String(CollectorZipkinHTTPHostPort, "", "The host:port (e.g. 127.0.0.1:9411 or :9411) of the collector's Zipkin server (disabled by default)")
```

**Review by yurishkuro — APPROVED** [review]

File: cmd/collector/app/builder_flags.go (hunk 4 above)
Comment (yurishkuro) [reply cmd/collector/app/builder_flags.go:104]: Fixing in #2554

**Comment by tobiaskohlbau** [comment]
Thanks for the review. I will add resolves on my next PR. Wasn't aware how the project handles this. At first I've included that line in my commit message but reverted that as I saw it's not used by the other commits in this project.

**Pull request merged**